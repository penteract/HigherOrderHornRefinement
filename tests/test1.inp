environment
add : int->int->int->bool
iter:(int->int->int->bool)->int->int->bool;

program
z = x + y ⇒ add x y z
n ≤ 0 ∧ r = 0 ⇒ iter f n r
( ∃ p : int. n > 0 ∧ iter f (n − 1 ) p ∧ f n p r ) ⇒ iter f n r;

goal
E n,r:int. iter add n r ^ n > r
