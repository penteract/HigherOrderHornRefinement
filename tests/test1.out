x_33 = x_31 + x_32 => x_9 x_33 x_32 x_31
x_24 x_30 x_29 x_28 => x_13 x_30 x_29 x_28
x_13 x_36 x_35 x_34 => x_24 x_36 x_35 x_34
x_37 <= 0 ^ x_38 = 0 \/ (E x_27:int.x_37 > 0 ^ x_16 x_27 (x_37 - 1) ^ x_24 x_38 x_27 x_37) => x_16 x_38 x_37
x_9 x_43 x_42 x_41 => x_13 x_43 x_42 x_41

goal:
E x_39:int.E x_40:int.x_16 x_40 x_39 ^ x_39 > x_40

